lib/uktt/parser/json_api.rb

```diff
@@ -0,0 +1,77 @@
+module Uktt
+  class Parser
+    class JsonApi
+      def initialize(body)
+        @body = JSON.parse(body)
+      end
+
+      def parse
+        if data.is_a?(Hash)
+          parse_resource(data)
+        elsif data.is_a?(Array)
+          parse_collection(data)
+        else
+          data
+        end
+      end
+
+      def errors
+        @body['error'] || @body['errors']&.map { |error| error['detail'] }&.join(', ')
+      end
+
+      private
+
+      def data
+        @data = @body['data']
```

Comment:
Think it's better to memoize here? If you look at https://github.com/TransformCore/uktt/pull/4/files#diff-519c672d196692b9521bfd813f86a464ddbadc6eeae52fd9b0b2e6cfdf8f35b7R9 you call data twice.

Reply:
Sounds good man